geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisString.java

```diff
@@ -718,15 +709,16 @@ protected void handleSetExpiration(SetOptions options) {
   ////// methods that modify the "value" field ////////////
 
   protected void valueAppend(byte[] bytes) {
-    value.append(bytes);
-  }
-
-  protected void valueSet(ByteArrayWrapper newValue) {
-    value = newValue;
+    int initialLength = value.length;
+    int additionalLength = bytes.length;
+    byte[] combined = new byte[initialLength + additionalLength];
+    System.arraycopy(value, 0, combined, 0, initialLength);
+    System.arraycopy(bytes, 0, combined, initialLength, additionalLength);
+    value = combined;
   }
 
-  protected void valueSetBytes(byte[] bytes) {
-    value.setBytes(bytes);
+  protected void valueSet(byte[] bytes) {
```

Comment:
This class uses a mix of this method and direct assignment to `value`. I'd prefer just one approach. Don't mind which, but would lean towards just direct `value = ...`